output/transform/merge_partition_sizes.go

```diff
@@ -12,11 +12,17 @@ import (
 //
 // TODO: recursively build up stats when nested partitioning is used
 func mergePartitionSizes(s snapshot.FullSnapshot, newState state.PersistedState, ts state.TransientState, databaseOidToIdx OidToIdx) snapshot.FullSnapshot {
+	relIdxToStatsIdx := make(map[int32]int, len(s.RelationStatistics))
+	for idx, stat := range s.RelationStatistics {
+		relIdxToStatsIdx[stat.RelationIdx] = idx
+	}
+
 	for idx, rel := range s.RelationInformations {
 		if !rel.HasParentRelation || rel.PartitionBoundary == "" {
 			continue
 		}
-		stat := s.RelationStatistics[idx]
+		statIdx := relIdxToStatsIdx[int32(idx)]
```

Comment:
Is there a reason the map couldn't use an `int` type to avoid this cast?

Reply:
Ah, but I suppose `stat.RelationIdx` is `int32` - that seems fine then.

Reply:
Protobuf makes `stat.RelationIdx` an `int32`, so I think we need a cast somewhere...

Reply:
Uhm, so reading PR description, let's say there is a case of:
* s.RelationInformations has the table tbl1 (idx: 3) information
* s.RelationStatistics doesn't have the table tb1 stats

Correct? In that case, `relIdxToStatsIdx` doesn't have something in `relIdxToStatsIdx[3]` and we should skip doing below in that case?

Reply:
Yeah, you're absolutely right: I'm building the map, but not using it correctly. Will fix.